.travis.yml

```diff
@@ -1,11 +1,34 @@
 language: node_js
-node_js:
-  - 8
+
+before_install:
+  # Export the C++11 compiler.
+  - if [[ $TRAVIS_OS_NAME == "linux" ]]; then export CXX=g++-4.8; fi
+
 cache:
   directories:
-    - node_modules
+    - $TRAVIS_BUILD_DIR/node_modules
+    - $HOME/.nvm
 # Note: travis defaults used for install
 # If custom install desired, include `npm install`
+
+# Do two runs, one for testing and one for linting.
+matrix:
+  include:
+  - node_js: 8
+    env: TEST_TYPE=test
+  - node_js: node 
+    env: TEST_TYPE=test:coverage
+  - node_js: node
+    env: TEST_TYPE=lint:eslint
+  - node_js: node
+    env: TEST_TYPE=lint:prettier
+  allow_failures:
+    - env: TEST_TYPE=lint:eslint
+    - env: TEST_TYPE=lint:prettier
```

Comment:
I believe this is the part that allows linting to fail and still call the build a success. This is a good thing.